provider/terraform/website/docs/version_2_upgrade.html.markdown

```diff
@@ -0,0 +1,176 @@
+
+Upgrade topics:
+
+<!-- TOC depthFrom:2 depthTo:2 -->
+
+- [Provider Version Configuration](#provider-version-configuration)
+- [Beta provider](#beta-provider)
+- [Open in Cloud Shell](#open-in-cloud-shell)
+- [Resource: google_bigtable_instance](#resource-google_bigtable_instance)
+- [Resource: google_compute_instance](#resource-google_compute_instance)
+- [Resource: google_compute_instance_from_template](#resource-google_compute_instance_from_template)
+- [Resource: google_compute_project_metadata](#resource-google_compute_project_metadata)
+- [Resource: google_compute_url_map](#resource-google_compute_url_map)
+- [Resource: google_project_iam_policy](#resource-google_project_iam_policy)
+- [Resource: google_sql_database_instance](#resource-google_sql_database_instance)
+- [Resource: google_storage_default_object_acl](#resource-google_storage_default_object_acl)
+- [Resource: google_storage_object_acl](#resource-google_storage_object_acl)
+
+<!-- /TOC -->
+
+## Provider Version Configuration
+
+!> **WARNING:** This topic is placeholder documentation until version 2.0.0 is released later this year.
+
+-> Before upgrading to version 2.0.0, it is recommended to upgrade to the most recent 1.X version of the provider and ensure that your environment successfully runs [`terraform plan`](https://www.terraform.io/docs/commands/plan.html) without unexpected changes or deprecation notices.
+
+It is recommended to use [version constraints when configuring Terraform providers](https://www.terraform.io/docs/configuration/providers.html#provider-versions). If you are following that recommendation, update the version constraints in your Terraform configuration and run [`terraform init`](https://www.terraform.io/docs/commands/init.html) to download the new version.
+
+For example, given this previous configuration:
+
+```hcl
+provider "google" {
+  # ... other configuration ...
+
+  version = "~> 1.19.0"
+}
+```
+
+An updated configuration:
+
+```hcl
+provider "google" {
+  # ... other configuration ...
+
+  version = "~> 2.0.0"
+}
+```
+
+## Beta provider
+
+The beta provider is now necessary to be able to configure resources with beta features.
+
+See [Provider Versions](https://terraform.io/docs/providers/google/provider_versions.html)
+for more details on how to use `google-beta`.
+
+## Open in Cloud Shell
+
+2.0.0 is the first release including Open in Cloud Shell. Examples in the documentation for
+Magic Modules resources now have Open in Cloud Shell links in their documentation that open
+them in an interactive editor and shell - all without leaving the browser. See the
+[blog post announcing the feature](https://www.hashicorp.com/blog/kickstart-terraform-on-gcp-with-google-cloud-shell)
+for more details.
+
+## Resource: google_bigtable_instance
```

Comment:
Nit: My personal preference is for references to resources, & fields to be in `code text` but this is going to have _a lot_ of references so I don't feel too strongly about it & feel free to ignore